tests/test_examples.py

```diff
@@ -236,118 +248,142 @@ def test_run_rbfe_legs(
+            seed=seed,
+            legs=leg,
+            n_eq_steps=n_eq_steps,
+            n_frames=n_frames,
+            n_windows=n_windows,
+            # Use simple charges to avoid os-dependent charge differences
+            forcefield="smirnoff_1_1_0_sc.py",
+            output_dir=f"{ARTIFACT_DIR_NAME}/rbfe_{mol_a}_{mol_b}_{leg}_{seed}",
+        )
+
+        def verify_run(output_dir: Path):
+            assert output_dir.is_dir()
+            assert (output_dir / "md_params.pkl").is_file()
+            assert (output_dir / "atom_mapping.svg").is_file()
+            assert (output_dir / "core.pkl").is_file()
+            assert (output_dir / "ff.py").is_file()
+
+            assert Forcefield.load_from_file(output_dir / "ff.py") is not None
+
+            leg_dir = output_dir / leg
+            assert leg_dir.is_dir()
+            assert (leg_dir / "results.npz").is_file()
+            assert (leg_dir / "lambda0_traj.npz").is_file()
+            assert (leg_dir / "lambda1_traj.npz").is_file()
+
+            assert (leg_dir / "simulation_result.pkl").is_file()
+            if leg in ["solvent", "complex"]:
+                assert (leg_dir / "host_config.pkl").is_file()
+            else:
+                assert not (leg_dir / "host_config.pkl").is_file()
+            assert (leg_dir / "hrex_transition_matrix.png").is_file()
+            assert (leg_dir / "hrex_swap_acceptance_rates_convergence.png").is_file()
+            assert (leg_dir / "hrex_replica_state_distribution_heatmap.png").is_file()
+
+            results = np.load(str(leg_dir / "results.npz"))
+            assert results["pred_dg"].size == 1
+            assert results["pred_dg"].dtype == np.float64
+            assert results["pred_dg"] != 0.0
+
+            assert results["pred_dg_err"].size == 1
+            assert results["pred_dg_err"].dtype == np.float64
+            assert results["pred_dg_err"] != 0.0
+
+            assert results["n_windows"].size == 1
+            assert results["n_windows"].dtype == np.intp
+            assert 2 <= results["n_windows"] <= config["n_windows"]
+            assert isinstance(results["overlaps"], np.ndarray)
+            assert all(isinstance(overlap, float) for overlap in results["overlaps"])
+
+            for lamb in [0, 1]:
+                traj_data = np.load(str(leg_dir / f"lambda{lamb:d}_traj.npz"))
+                assert len(traj_data["coords"]) == n_frames
+                assert len(traj_data["boxes"]) == n_frames
+
+        def verify_endstate_hashes(output_dir: Path):
+            leg_dir = output_dir / leg
+            endstate_0_hash = hash_file(leg_dir / "lambda0_traj.npz")
+            endstate_1_hash = hash_file(leg_dir / "lambda1_traj.npz")
+            assert endstate_0_hash == endstate_hashes[leg][0] and endstate_1_hash == endstate_hashes[leg][1], (
+                f"{endstate_0_hash} != {endstate_hashes[leg][0]} and/or {endstate_1_hash} != {endstate_hashes[leg][1]}"
```

Comment:
nit: probably don't need to repeat the hashes in the assert message, since pytest should print the diff anyway?

Reply:
Ah, I see, makes sense to avoid the need to run the test twice to update both hashes.

I suppose comparing tuples in the assertion would also print both hashes as desired?

```python
assert (endstate_0_hash, endstate_1_hash) == (endstate_hashes[leg][0], endstate_hashes[leg][1])
```